File: frontend/bibliotecaArcadia/bibliotecaArcadia/10 - Modelos/Notificacion.swift
//
//  Notificacion.swift
//  bibliotecaArcadia
//
//  Created by Sandra MZ on 17/4/25.
//

import Foundation

struct Notificacion: Decodable {
    let id: Int
    let mensaje: String
    let fecha: String
    let portada: String?
    let id_usuario: Int
    let id_libro: String
}

File: backend/api_arcadia/app/Models/Notificacion.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Notificacion extends Model
{
    protected $table = 'notificaciones';

    protected $fillable = [
        'mensaje',
        'fecha',
        'id_usuario',
        'id_libro',
        'portada'
    ];

    protected function casts(): array
    {
        return [
            'fecha' => 'date'
        ];
    }

    public function usuario(): BelongsTo{
        return $this->belongsTo(Usuario::class, 'id_usuario');
    }
    public function libro(): BelongsTo{
        return $this->belongsTo(Libro::class, 'id_libro');
    }
}
